sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -21,7 +21,8 @@ public virtual bool IsRetriableResponse(HttpMessage message)
         /// </summary>
         public virtual bool IsRetriableException(Exception exception)
         {
-            return (exception is IOException);
+            return (exception is IOException) ||
+                   (exception is RequestFailedException requestFailed && requestFailed.Status == 0);
```

Comment:
This might be a separate issue, but should certain HTTP status codes be retried (like `503 Service Unavailable`)?

Reply:
Retry policy is part of the pipeline and gets to observe responses before they are turned into exceptions. The logic to decide if response is retryable is in https://github.com/Azure/azure-sdk-for-net/blob/61c8721cde56ec1fa9273c9cf872413643ab4298/sdk/core/Azure.Core/src/ResponseClassifier.cs#L14

There are two reasons why retry policy might be catching `RequestFailedException`:
1. If we weren't even able to get to the server and read the response. And this is what we handling with this PR.
2. Some policy down the pipeline decided to turn the response into an exception prematurely and interrupt the pipeline flow. This one we shouldn't handle and let bubble up.